pkg/cloud/azureprovider.go

```diff
@@ -734,6 +828,49 @@ func (az *Azure) AllNodePricing() (interface{}, error) {
 func (az *Azure) NodePricing(key Key) (*Node, error) {
 	az.DownloadPricingDataLock.RLock()
 	defer az.DownloadPricingDataLock.RUnlock()
+	config, _ := az.Config.GetCustomPricingData()
+	if config.SpotLabel != "" && config.SpotLabelValue != "" {
+		labelsMapStr := fmt.Sprintf("%s", key)
+		spotLabelMapStr := " " + config.SpotLabel + ":" + config.SpotLabelValue + " "
```

Comment:
Padding is there, because `labelsMapStr` will have all the labels separated by space, and `substr` can find needed string even in the middle of another string (this is kind of just for safety, because you can set spot labels while deploying cluster and it could be something really simple, which could be already there within another label, like: `kubernetes.azure.com/scalesetpriority: mylabel` and `kubernetes.azure.com/scalesetpriority: mylabelspot`).
> but why not use fmt.Sprintf(" %s:%s ", config.SpotLabel, config.SpotLabelValue) since you use it right above

To be honest I don't know why I did it, there was some explanation, but I can't remember it now. So I will change that.
The thing is, that I've done those changes long-long ago, but only recently Microsoft has added a way to query different currencies (before that prices were in USD only), which we was waiting for. So I might miss something that I was doing back then.

Reply:
Makes sense, if you don't mind changing this one I would appreciate it.